### PR TITLE
fix: support interlining for activeTripId in vehicles-for-agency

### LINKED ISSUE #1131: [spec-gap] vehicles-for-agency: activeTripId ignores interlining
### Description:
Per Spec Extension 5b (maglev.wiki/vehicles-for-agency.md), when a vehicle is interlining, the activeTripId (the trip actually being executed) may differ from the outer tripId (the nominal trip). In this scenario, both trips must be included in references.trips.

Currently, in vehicles_for_agency_handler.go, activeTripId is blindly set to equal the outer tripId. Interlining is never represented, and the actual active trip is not pushed to the references.

### Expected behavior:

* activeTripId reflects the actual executing trip, which may differ from the outer tripId during interlining.

* Both the nominal trip and the active trip are appended to references.trips when they differ.

### Actual behavior:

* activeTripId always equals the outer tripId.

* Interlining is completely unhandled.

### PR DESCRIPTION
**Pull request by 3rabiii**
### Summary

Aligns the `vehicles-for-agency` endpoint with spec Extension 5b (interlining).
When a vehicle is interlining, `tripStatus.activeTripId` — the trip actually being
executed — may differ from the outer `tripId` (the nominal GTFS-RT trip). Both
trips are now included in `references.trips`.

Previously `activeTripId` was set equal to the outer `tripId` unconditionally, so
interlining was never represented and the active trip was missing from references.

### Changes

- **`vehicles_helper.go`** — Add `resolveActiveTripID`, which resolves the trip
  executing at the reference time within the vehicle's block by matching the block
  trip whose scheduled window contains the reference time-of-day. Falls back to the
  nominal trip when there is no block, no active services, or no window match.
- **`vehicles_for_agency_handler.go`** — Use `resolveActiveTripID` for
  `tripStatus.activeTripId`, and append the active trip to `references.trips` (with
  its own route id) when it differs from the nominal trip.

### Testing

- `TestResolveActiveTripID` — unit coverage for the resolver: interlining match,
  nominal-window match, no-window fallback, and unknown-trip fallback.
- `TestVehiclesForAgencyHandler_InterliningActiveTrip` — discovers a real
  interlining scenario in the RABA fixture and asserts `activeTripId` differs from
  the outer `tripId` and that both trips appear in `references.trips`.

Closes: #1131 

## Summary by CodeRabbit

- **Bug Fixes**
  - Vehicle status and `/vehicles-for-agency` now correctly resolve and report the actively executing trip at the requested reference time, including interlined scenarios.
  - Responses preserve the nominal trip ID while adding the resolved active trip plus the correct trip/route references.
  - Active-trip resolution now uses DST-stable “wall clock” timing and handles scheduled windows that extend past midnight with safer fallbacks.
- **Tests**
  - Added integration and unit test coverage for interlining and active-trip resolution edge cases, including DST and >24:00 windows.

### CODE REVIEW

**Comment by 3rabiii** [comment]
@coderabbitai full review

File: internal/restapi/vehicles_helper.go [hunk 1]
```diff
@@ -159,6 +161,60 @@ func GetVehicleActiveTripID(vehicle *gtfs.Vehicle) string {
...
+
+	// Check the current service day, then the previous day for trips whose windows
+	// run past 24:00 (GTFS allows e.g. 25:30:00). A trip belonging to the previous
+	// service day is matched against referenceTime offset by +24h.
+	if tripID, ok := api.activeTripInBlockAt(ctx, nominalTrip.BlockID, referenceTime, sinceMidnightNs); ok {
+		return tripID
+	}
+	prevDay := referenceTime.AddDate(0, 0, -1)
+	if tripID, ok := api.activeTripInBlockAt(ctx, nominalTrip.BlockID, prevDay, sinceMidnightNs+int64(24*time.Hour)); ok {
+		return tripID
```
[the remark this answers is not in the document]
Comment (3rabiii) [reply internal/restapi/vehicles_helper.go]: I"m working on this @burma-shave
